operations/mimir-mixin/alerts/blocks.libsonnet

```diff
@@ -9,9 +9,9 @@
           alert: $.alertName('IngesterHasNotShippedBlocks'),
           'for': '15m',
           expr: |||
-            (min by(%(alert_aggregation_labels)s, %(per_instance_label)s) (time() - thanos_objstore_bucket_last_successful_upload_time{%(per_job_label)s=~".+/ingester.*"}) > 60 * 60 * 4)
+            (min by(%(alert_aggregation_labels)s, %(per_instance_label)s) (time() - thanos_objstore_bucket_last_successful_upload_time{%(job_regex)s}) > 60 * 60 * 4)
             and
-            (max by(%(alert_aggregation_labels)s, %(per_instance_label)s) (thanos_objstore_bucket_last_successful_upload_time{%(per_job_label)s=~".+/ingester.*"}) > 0)
+            (max by(%(alert_aggregation_labels)s, %(per_instance_label)s) (thanos_objstore_bucket_last_successful_upload_time{%(job_regex)s}) > 0)
```

Comment:
This approach is a bit fragile. Let's look at the resulting query:
```
(min by(cluster, namespace, pod) (time() - thanos_objstore_bucket_last_successful_upload_time{job=~".*/(ingester.*|cortex|mimir|mimir-write.*)"}) > 60 * 60 * 4)
and
(max by(cluster, namespace, pod) (thanos_objstore_bucket_last_successful_upload_time{job=~".*/(ingester.*|cortex|mimir|mimir-write.*)"}) > 0)
```

The alerts uses a very generic metric `thanos_objstore_bucket_last_successful_upload_time ` which gets updated each time an object to uploaded to the storage. It's not necessary a TSDB block upload.

With the microservices deployment, we assumed it was always a block because the ingester can only upload it, but the new regex works for monolithic mode as well, and it could be any upload, not just a TSDB block.

I think a better way to address it would be introduce a new metric specific to the shipper, with the last time a block was successfully uploaded by the shipper. Then the alert can be based on that new metric.

Reply:
Taken a pass at this. Need to add tests but let me know what you think before I do.